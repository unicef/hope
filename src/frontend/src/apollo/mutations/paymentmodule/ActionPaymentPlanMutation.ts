import { gql } from '@apollo/client';

export const ACTION_PAYMENT_PLAN = gql`
  mutation ActionPP($input: ActionPaymentPlanInput!) {
    actionPaymentPlanMutation(input: $input) {
      paymentPlan {
        id
        status
      }
    }
  }
`;
