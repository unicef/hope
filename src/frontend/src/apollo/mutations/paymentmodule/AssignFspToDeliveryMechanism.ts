import { gql } from '@apollo/client';

export const ASSIGN_FSP_TO_DELIVERY_MECHANISM = gql`
  mutation AssignFspToDeliveryMech($input: AssignFspToDeliveryMechanismInput!) {
    assignFspToDeliveryMechanism(input: $input) {
      paymentPlan {
        id
        deliveryMechanisms {
          id
          name
          fsp {
            id
            name
          }
        }
      }
    }
  }
`;
