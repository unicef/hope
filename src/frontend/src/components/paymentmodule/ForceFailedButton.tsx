import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogContainer } from '@containers/dialogs/DialogContainer';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { useMarkPayAsFailedMutation } from '@generated/graphql';

export interface ForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function ForceFailedButton({
  paymentId,
  disabled = false,
}: ForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useMarkPayAsFailedMutation();

  const submit = async (): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
        },
      });
      setOpenModal(false);
      showMessage(t('Payment has been marked as failed.'));
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-mark-as-failed"
          disabled={disabled}
        >
          {t('Mark as failed')}
        </Button>
      </Box>
      <Dialog
        open={isOpenModal}
        onClose={() => setOpenModal(false)}
        scroll="paper"
        aria-labelledby="form-dialog-title"
        maxWidth="md"
      >
        <DialogTitleWrapper>
          <DialogTitle>{t('Mark as failed')}</DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <DialogContainer>
            <Box p={5}>
              {t('Are you sure you would like to mark payment as failed?')}
            </Box>
          </DialogContainer>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setOpenModal(false)}>{t('CANCEL')}</Button>
            <Button
              type="submit"
              color="primary"
              variant="contained"
              onClick={() => submit()}
              data-cy="button-submit"
              disabled={loading}
            >
              {t('Mark as failed')}
            </Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </Box>
  );
}
