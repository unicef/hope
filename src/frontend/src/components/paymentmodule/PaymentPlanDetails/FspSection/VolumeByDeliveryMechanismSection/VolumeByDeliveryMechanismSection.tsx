import { Box, Grid, Typography } from '@mui/material';
import * as React from 'react';
import { Pie } from 'react-chartjs-2';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { PaymentPlanQuery } from '@generated/graphql';
import { LabelizedField } from '@core/LabelizedField';
import { FieldBorder } from '@core/FieldBorder';
import type { ChartData, ChartOptions } from 'chart.js';

const Title = styled.div`
  padding-bottom: ${({ theme }) => theme.spacing(2)};
`;

const ContentWrapper = styled.div`
  display: flex;
`;

const ChartContainer = styled.div`
  width: 100px;
  height: 100px;
  margin: 0 auto;
`;

interface VolumeByDeliveryMechanismSectionProps {
  paymentPlan: PaymentPlanQuery['paymentPlan'];
}

const DeliveryMechanismsColorsMap = new Map([
  ['Cardless cash withdrawal', '#FC942A'],
  ['Cash', '#D8E1EE'],
  ['Cash by FSP', '#D8E1EE'],
  ['Cheque', '#10CB16'],
  ['Deposit to Card', '#4E606A'],
  ['In Kind', ' #d8d8d8'],
  ['Mobile Money', '#e4e4e4'],
  ['Other', '#EF4343'],
  ['Pre-paid card', '#D9D1CE'],
  ['Referral', '#715247'],
  ['Transfer', '#003C8F'],
  ['Transfer to Account', '#003C8F'],
  ['Voucher', '#00ADEF'],
]);

export const getDeliveryMechanismColor = (
  deliveryMechanism: string,
): string => {
  if (DeliveryMechanismsColorsMap.has(deliveryMechanism)) {
    return DeliveryMechanismsColorsMap.get(deliveryMechanism);
  }
  return '#CCC';
};

export const VolumeByDeliveryMechanismSection: React.FC<
  VolumeByDeliveryMechanismSectionProps
> = ({ paymentPlan }) => {
  const { t } = useTranslation();
  const { volumeByDeliveryMechanism } = paymentPlan;

  const mappedDeliveryMechanism = volumeByDeliveryMechanism.map(
    (vdm, index) => (
      <Grid
        item
        xs={6}
        /* eslint-disable-next-line react/no-array-index-key */
        key={`${vdm.deliveryMechanism.id}-${index}`}
      >
        <FieldBorder
          color={getDeliveryMechanismColor(vdm.deliveryMechanism.name)}
        >
          <LabelizedField
            label={`${vdm.deliveryMechanism.name} (${vdm.deliveryMechanism.fsp?.name})`}
            value={`${vdm.volume} ${paymentPlan.currency} (${vdm.volumeUsd} USD)`}
          />
        </FieldBorder>
      </Grid>
    ),
  );

  const chartLabels = volumeByDeliveryMechanism.map(
    (el) => `${el.deliveryMechanism.name} (${el.deliveryMechanism.fsp?.name})`,
  );

  const chartData = volumeByDeliveryMechanism.map((el) => el.volumeUsd);

  const chartColors = (): string[] => {
    return volumeByDeliveryMechanism.map((el) =>
      getDeliveryMechanismColor(el.deliveryMechanism.name),
    );
  };

  const data: ChartData<'pie'> = {
    labels: chartLabels,
    datasets: [
      {
        data: chartData,
        backgroundColor: chartColors(),
      },
    ],
  };

  const options: ChartOptions<'pie'> = {
    plugins: {
      legend: {
        display: false,
      },
    },
  };

  return (
    <Box display="flex" flexDirection="column">
      <Title>
        <Typography variant="h6">
          {t('Volume by Delivery Mechanism')}
        </Typography>{' '}
      </Title>
      <ContentWrapper>
        <Grid container spacing={0} justifyContent="flex-start">
          {mappedDeliveryMechanism}
        </Grid>
        <Grid
          container
          spacing={0}
          justifyContent="flex-start"
          alignItems="center"
        >
          <Grid item xs={4}>
            <ChartContainer>
              <Pie data={data} options={options} />
            </ChartContainer>
          </Grid>
        </Grid>
      </ContentWrapper>
    </Box>
  );
};
