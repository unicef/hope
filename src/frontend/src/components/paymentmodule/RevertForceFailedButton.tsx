import { Box, Button, DialogContent, DialogTitle } from '@mui/material';
import * as React from 'react';
import { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Field, Form, Formik } from 'formik';
import * as Yup from 'yup';
import CalendarTodayRoundedIcon from '@mui/icons-material/CalendarTodayRounded';
import { Dialog } from '@containers/dialogs/Dialog';
import { DialogActions } from '@containers/dialogs/DialogActions';
import { DialogFooter } from '@containers/dialogs/DialogFooter';
import { DialogTitleWrapper } from '@containers/dialogs/DialogTitleWrapper';
import { useSnackbar } from '@hooks/useSnackBar';
import { FormikTextField } from '@shared/Formik/FormikTextField';
import { FormikDateField } from '@shared/Formik/FormikDateField';
import { useRevertMarkPayAsFailedMutation } from '@generated/graphql';

export interface RevertForceFailedButtonProps {
  paymentId: string;
  disabled?: boolean;
}
export function RevertForceFailedButton({
  paymentId,
  disabled = false,
}: RevertForceFailedButtonProps): React.ReactElement {
  const { t } = useTranslation();
  const [isOpenModal, setOpenModal] = useState(false);
  const { showMessage } = useSnackbar();
  const [mutate, { loading }] = useRevertMarkPayAsFailedMutation();

  const validationSchema = Yup.object().shape({
    deliveredQuantity: Yup.number()
      .min(0)
      .max(99999999, t('Number is too big')),
    deliveryDate: Yup.date().required(t('Delivery date is required')),
  });

  const submit = async (values, { resetForm }): Promise<void> => {
    try {
      await mutate({
        variables: {
          paymentId,
          deliveredQuantity: values.deliveredQuantity,
          deliveryDate: values.deliveryDate,
        },
      });
      setOpenModal(false);
      showMessage(t('Force failed has been reverted.'));
      resetForm();
    } catch (e) {
      e.graphQLErrors.map((x) => showMessage(x.message));
    }
  };

  return (
    <Box>
      <Box p={2}>
        <Button
          color="primary"
          variant="contained"
          onClick={() => setOpenModal(true)}
          data-cy="button-revert-mark-as-failed"
          disabled={disabled}
        >
          {t('Revert mark as failed')}
        </Button>
      </Box>
      <Formik
        initialValues={{
          deliveredQuantity: 0,
          deliveryDate: '',
        }}
        validationSchema={validationSchema}
        onSubmit={submit}
      >
        {({ submitForm, resetForm }) => (
          <Dialog
            open={isOpenModal}
            onClose={() => setOpenModal(false)}
            scroll="paper"
            aria-labelledby="form-dialog-title"
            maxWidth="md"
          >
            <DialogTitleWrapper>
              <DialogTitle>{t('Revert mark as failed')}</DialogTitle>
            </DialogTitleWrapper>
            <DialogContent>
              <Form>
                <Field
                  name="deliveredQuantity"
                  label={t('Delivered Quantity')}
                  type="number"
                  required
                  fullWidth
                  variant="outlined"
                  component={FormikTextField}
                />
                <Field
                  name="deliveryDate"
                  label={t('Delivery Date')}
                  component={FormikDateField}
                  required
                  fullWidth
                  decoratorEnd={<CalendarTodayRoundedIcon color="disabled" />}
                />
              </Form>
            </DialogContent>
            <DialogFooter>
              <DialogActions>
                <Button
                  onClick={() => {
                    setOpenModal(false);
                    resetForm();
                  }}
                >
                  {t('CANCEL')}
                </Button>
                <Button
                  type="submit"
                  color="primary"
                  variant="contained"
                  onClick={submitForm}
                  data-cy="button-submit"
                  disabled={loading}
                >
                  {t('Revert mark as failed')}
                </Button>
              </DialogActions>
            </DialogFooter>
          </Dialog>
        )}
      </Formik>
    </Box>
  );
}
