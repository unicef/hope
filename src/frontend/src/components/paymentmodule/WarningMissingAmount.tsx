import { Box } from '@mui/material';
import WarningIcon from '@mui/icons-material/Warning';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { ReactElement } from 'react';

const WarnIcon = styled(WarningIcon)`
  color: ${({ theme }) => theme.hctPalette.orange};
`;

interface WarningMissingAmountProps {
  amount: number;
  currency: string;
}

export function WarningMissingAmount({
  amount,
  currency,
}: WarningMissingAmountProps): ReactElement {
  const { t } = useTranslation();

  return (
    <Box display="flex" alignItems="center">
      <WarnIcon />
      {t('Missing')} {amount} {currency}
    </Box>
  );
}
