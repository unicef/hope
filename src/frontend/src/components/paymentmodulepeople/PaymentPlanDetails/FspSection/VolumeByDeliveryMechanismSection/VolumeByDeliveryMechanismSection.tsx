import { Box, Grid2 as Grid, Typography } from '@mui/material';
import { Pie } from 'react-chartjs-2';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { LabelizedField } from '@core/LabelizedField';
import { FieldBorder } from '@core/FieldBorder';
import type { ChartData, ChartOptions } from 'chart.js';
import { FC } from 'react';
import { PaymentPlanDetail } from '@restgenerated/models/PaymentPlanDetail';

const Title = styled.div`
  padding-bottom: ${({ theme }) => theme.spacing(2)};
`;

const ContentWrapper = styled.div`
  display: flex;
`;

const ChartContainer = styled.div`
  width: 100px;
  height: 100px;
  margin: 0 auto;
`;

interface VolumeByDeliveryMechanismSectionProps {
  paymentPlan: PaymentPlanDetail;
}

const DeliveryMechanismsColorsMap = new Map([
  ['Cardless cash withdrawal', '#FC942A'],
  ['Cash', '#D8E1EE'],
  ['Cash by FSP', '#D8E1EE'],
  ['Cheque', '#10CB16'],
  ['Deposit to Card', '#4E606A'],
  ['In Kind', ' #d8d8d8'],
  ['Mobile Money', '#e4e4e4'],
  ['Other', '#EF4343'],
  ['Pre-paid card', '#D9D1CE'],
  ['Referral', '#715247'],
  ['Transfer', '#003C8F'],
  ['Transfer to Account', '#003C8F'],
  ['Voucher', '#00ADEF'],
]);

export const getDeliveryMechanismColor = (
  deliveryMechanism: string,
): string => {
  if (DeliveryMechanismsColorsMap.has(deliveryMechanism)) {
    return DeliveryMechanismsColorsMap.get(deliveryMechanism);
  }
  return '#CCC';
};

export const VolumeByDeliveryMechanismSection: FC<
  VolumeByDeliveryMechanismSectionProps
> = ({ paymentPlan }) => {
  const { t } = useTranslation();
  const { volume_by_delivery_mechanism } = paymentPlan;

  const mappedDeliveryMechanism = volume_by_delivery_mechanism?.map(
    (vdm, index) => (
      <Grid
        size={{ xs: 6 }}
        /* eslint-disable-next-line react/no-array-index-key */
        key={`${vdm.delivery_mechanism.id}-${index}`}
      >
        <FieldBorder
          color={getDeliveryMechanismColor(vdm.delivery_mechanism.name)}
        >
          <LabelizedField
            label={`${vdm.delivery_mechanism.name} (${vdm.delivery_mechanism.fsp?.name ?? '-'})`}
            value={`${vdm.volume ?? '0.00'} ${paymentPlan.currency} (${vdm.volume_usd ?? '0.00'} USD)`}
          />
        </FieldBorder>
      </Grid>
    ),
  );

  const chartLabels = volume_by_delivery_mechanism.map(
    (el) => `${el.deliveryMechanism.name} (${el.deliveryMechanism.fsp?.name})`,
  );

  const chartData = volume_by_delivery_mechanism.map((el) => el.volumeUsd);

  const chartColors = (): string[] => {
    return volume_by_delivery_mechanism.map((el) =>
      getDeliveryMechanismColor(el.deliveryMechanism.name),
    );
  };

  const data: ChartData<'pie'> = {
    labels: chartLabels,
    datasets: [
      {
        data: chartData,
        backgroundColor: chartColors(),
      },
    ],
  };

  const options: ChartOptions<'pie'> = {
    plugins: {
      legend: {
        display: false,
      },
    },
  };

  return (
    <Box display="flex" flexDirection="column">
      <Title>
        <Typography variant="h6">
          {t('Volume by Delivery Mechanism')}
        </Typography>{' '}
      </Title>
      <ContentWrapper>
        <Grid container spacing={0} justifyContent="flex-start">
          {mappedDeliveryMechanism}
        </Grid>
        <Grid
          container
          spacing={0}
          justifyContent="flex-start"
          alignItems="center"
        >
          <Grid size={{ xs: 4 }}>
            <ChartContainer>
              <Pie data={data} options={options} />
            </ChartContainer>
          </Grid>
        </Grid>
      </ContentWrapper>
    </Box>
  );
};
