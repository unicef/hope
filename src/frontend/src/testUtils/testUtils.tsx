import { ReactElement } from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, RenderOptions } from '@testing-library/react';
import { Formik } from 'formik';
import noop from 'lodash/noop';
import { TestProviders } from './testProviders';
import { ProgramStatus } from '@generated/graphql';
import { MockLink } from '@apollo/client/testing';

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
) =>
  render(
    <>
      <BrowserRouter>
        <Formik initialValues={{}} onSubmit={noop}>
          {ui}
        </Formik>
      </BrowserRouter>
    </>,
    {
      wrapper: TestProviders,
      ...options,
    },
  );

export * from '@testing-library/react';
export { customRender as render };

let m_w = 123456789;
let m_z = 987654321;
const mask = 0xffffffff;

// Takes any integer
export const seed = (i) => {
  m_w = (123456789 + i) & mask;
  m_z = (987654321 - i) & mask;
};

// Returns number between 0 (inclusive) and 1.0 (exclusive),
// just like Math.random().
export const random = () => {
  m_z = (36969 * (m_z & 65535) + (m_z >> 16)) & mask;
  m_w = (18000 * (m_w & 65535) + (m_w >> 16)) & mask;
  let result = ((m_z << 16) + (m_w & 65535)) >>> 0;
  result /= 4294967296;
  return result;
};

export class ApolloLoadingLink extends MockLink {
  constructor() {
    super([]);
  }

  request() {
    return null;
  }
}

export const fakeContextProgram = {
  selectedProgram: {
    id: '1',
    name: 'someName',
    status: ProgramStatus.Active,
    pduFields: null,
    dataCollectingType: {
      id: '1',
      householdFiltersAvailable: true,
      individualFiltersAvailable: true,
      label: 'data collecting type',
      code: '123',
      type: 'full',
      children: null,
    },
  },
  setSelectedProgram: () => {},
  isActiveProgram: true,
  isSocialDctType: false,
  isStandardDctType: true,
  programHasPdu: false,
};
