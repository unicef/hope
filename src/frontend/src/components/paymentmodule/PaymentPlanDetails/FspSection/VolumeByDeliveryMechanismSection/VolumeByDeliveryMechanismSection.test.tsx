import * as React from 'react';
import { fakeApolloPaymentPlan } from '../../../../../../fixtures/paymentmodule/fakeApolloPaymentPlan';
import { render } from '../../../../../testUtils/testUtils';
import { VolumeByDeliveryMechanismSection } from './VolumeByDeliveryMechanismSection';

describe('components/paymentmodule/PaymentPlanDetails/VolumeByDeliveryMechanismSection', () => {
  it('should render', () => {
    const { container } = render(
      <VolumeByDeliveryMechanismSection paymentPlan={fakeApolloPaymentPlan} />,
    );
    expect(container).toMatchSnapshot();
  });
});
