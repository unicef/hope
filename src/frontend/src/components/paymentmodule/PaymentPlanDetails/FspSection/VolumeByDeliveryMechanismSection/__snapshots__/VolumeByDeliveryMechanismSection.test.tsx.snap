// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/paymentmodule/PaymentPlanDetails/VolumeByDeliveryMechanismSection should render 1`] = `
<div>
  <div
    class="MuiBox-root css-j7qwjs"
  >
    <div
      class="sc-hLQSwg IDXfo"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-1j53xw6-MuiTypography-root"
      >
        Volume by Delivery Mechanism
      </h6>
       
    </div>
    <div
      class="sc-eDLKkx eHrULA"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-1bzpx2o-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="sc-dmyCSP cjNvSp"
            color="#D8E1EE"
          >
            <div>
              <span
                class="sc-beySPh gbuZtJ"
                color="textSecondary"
              >
                Cash (Miranda Ltd)
              </span>
              <div
                data-cy="label-Cash (Miranda Ltd)"
              >
                <span
                  class="sc-guDLey cyWTlF"
                  color="textSecondary"
                >
                  0 PLN (0 USD)
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="sc-dmyCSP UzTpF"
            color=" #d8d8d8"
          >
            <div>
              <span
                class="sc-beySPh gbuZtJ"
                color="textSecondary"
              >
                In Kind (Frazier-Watson)
              </span>
              <div
                data-cy="label-In Kind (Frazier-Watson)"
              >
                <span
                  class="sc-guDLey cyWTlF"
                  color="textSecondary"
                >
                  0 PLN (0 USD)
                </span>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <div
            class="sc-dmyCSP bOKfMH"
            color="#003C8F"
          >
            <div>
              <span
                class="sc-beySPh gbuZtJ"
                color="textSecondary"
              >
                Transfer (Bray Group)
              </span>
              <div
                data-cy="label-Transfer (Bray Group)"
              >
                <span
                  class="sc-guDLey cyWTlF"
                  color="textSecondary"
                >
                  0 PLN (0 USD)
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-container css-1e2bu2o-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
        >
          <div
            class="sc-jTQCzO LEIUV"
          >
            <canvas
              height="0"
              role="img"
              style="display: block; box-sizing: border-box; height: 0px; width: 0px;"
              width="0"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
